### PR TITLE
Proposal forms inactive option

### PR DESCRIPTION
**Pull request by stuartjennings-up**
https://trello.com/c/lfsjepRm/218-8-form-updates

Happy to give more detail in person about what this card entails.

It tackles the problem that editing form fields meant that existing proposals or reviews that used the form lost data as the fields were edited. There was also a card for making forms active/inactive, and my solution involved a fix for that as well.

Essentially, it does the following:

- Combines the two almost identical systems for editing review forms and proposal forms into one.
- When forms are edited it creates a new form marked as `in_edit` that the changes apply to, that can then be saved.
- Forms can be previewed and marked as active or inactive from the form list page.
- Only active forms appear in the relevant workflows.

### CODE REVIEW

File: src/submission/forms.py [hunk 1]
```diff
@@ -1,7 +1,11 @@
 from django import forms
 from django.forms import ModelForm
 
-from core import models as core_models, logic as core_logic
+from core import (
```
Comment (terrasea2) [inline src/submission/forms.py:4]: Like how you've done this import. Very easy to read

Comment (stuartjennings-up) [reply src/submission/forms.py:4]: 👍

File: src/templates/manager/edit_form.html [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                    <a href="{% url 'manager_edit_form_element' form_type form.id field.field.id %}">
+                        Edit
+                    </a>
+                </td>
+				<td>
+                    <a href="{% url 'manager_delete_form_element' form_type form.id field.field.id %}">
+                        Delete
+                    </a>
+                </td>
+				{% ifequal field.above -1 %}
```
Comment (terrasea2) [inline src/templates/manager/edit_form.html]: Some indentation problems here

```{% ifequal field.above -1 %}``` should not be indented so far in, etc

File: src/templates/manager/forms.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+				<th>Form Name</th>
+				<th>Ref</th>
+				<th>Field Count</th>
+				<th></th>
+                <th></th>
+                <th></th>
+			</tr>
+		</thead>
+		<tbody>
+			{% for form in forms %}
```
Comment (terrasea2) [inline src/templates/manager/forms.html]: ```<tr>``` should be indented to be in the ```{% for form in forms %}``` as it makes templates easier to read if you do

File: src/templates/manager/preview_form.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                  {% else %}
+                    col-md-12
+                  {% endif %}">
+          {% if field.field.widget.attrs.is_checkbox %}
+            {% if field.errors %}
+                <span class="red">{{ field.errors|join:", " }}</span>
+                <br />
+            {% endif %}
+            {{ field }} &nbsp;
+              {% if field.field.help_text %}
```
Comment (terrasea2) [inline src/templates/manager/preview_form.html]: Unindent one for these. should be same indentation as ```{{ field }} &nbsp;```

File: src/templates/manager/preview_form.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+              {% endif %}
+          {% else %}
+             {% if field.errors %}
+                 <span class="red">{{ field.errors|join:", " }}</span>
+             {% endif %}
+             {% bootstrap_field field %}
+          {% endif %}
+      </div>
+  {% endfor %}
+</form>
```
Comment (terrasea2) [inline src/templates/manager/preview_form.html]: ```</form>``` end tag should be one more indentation in from the end tag ```</div>``` below it. I suspect you're problems arrise from the previous developers who didn't believe in formatting template code

File: src/templates/manager/preview_form.html [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+</div>  
+{% endblock body %}
```
Comment (terrasea2) [inline src/templates/manager/preview_form.html]: Same here, all stuff between ```{% block %}``` and ```{% endblock %}``` should be indented by one more than the block template tags

File: src/templates/manager/preview_form.html [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+{% if form_type == 'proposal' %}
```
Comment (terrasea2) [inline src/templates/manager/preview_form.html]: Still not perfect, but that's heading in the right direction.

File: src/templates/manager/preview_form.html [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    {% block js %}
+        <script type="text/javascript" src="{% static "js/fancyfile.js" %}"></script>
+        <script type="text/javascript">$("textarea").jqte();</script>
+        <script language="javascript" type="text/javascript">
+        $(document).ready(function () {
+               {%for field in default_fields%}
+                  {% if field.field.required %}
+                 var required_text=$('label[for="' + "id_{{field.name}}"+ '"]').html()+" *";
```
Comment (terrasea2) [inline src/templates/manager/preview_form.html]: indent

**Review by terrasea2 — CHANGES_REQUESTED** [review]

**Review by terrasea2 — APPROVED** [review]

**Comment by fradeve** [comment]
Is this good to go?

**Comment by stuartjennings-up** [comment]
@fradeve Nearly- Tim needs to test this final fix I've made then we can deploy. He's also asked for a small extra feature which won't take v long but if you're keen to get this out then we can go ahead without it.

**Pull request merged**